### PR TITLE
CI:update integration test with deployment check and push action enha…

### PR DESCRIPTION
**Pull request by SamYuan1990**
CI:update integration test with deployment check and push action enhancement

related with #192
related with #254 

Signed-off-by: Sam Yuan <yy19902439@126.com>

### CODE REVIEW

**Comment by SamYuan1990** [comment]
https://github.com/sustainable-computing-io/kepler/actions/runs/3152440537/jobs/5127657495#step:6:242
CI round test waiting looks good when success.

**Review by rootfs — APPROVED** [review]

**Comment by marceloamaral** [comment]
@SamYuan1990 looks like the integration test is failing now, see [here](https://github.com/sustainable-computing-io/kepler/actions/runs/3155912869/jobs/5135076208)

**Comment by SamYuan1990** [comment]
> @SamYuan1990 looks like the integration test is failing now, see [here](https://github.com/sustainable-computing-io/kepler/actions/runs/3155912869/jobs/5135076208)

I will try to have a fix.
the command with exit code 2 breaks the loop and it is not been catched.

File: hack/cluster-deploy.sh [hunk 1]
```diff
@@ -31,10 +31,29 @@ function main() {
...
     kubectl apply -f ${MANIFESTS_OUT_DIR} || true
-    kubectl rollout status daemonset kepler-exporter -n kepler --timeout 60s
-
-    echo "Check the logs of the kepler-exporter with:"
-    echo "kubectl -n kepler logs daemonset.apps/kepler-exporter"
+    
+    # round for 3 times and each for 60s
+    # check if the rollout status is running
+    deploy_status=1
+    for i in 1 2 3
```
Comment (marceloamaral) [inline hack/cluster-deploy.sh:38]: @SamYuan1990 instead of check this 3 times, why not use a larger timeout?
180s?

File: hack/cluster-deploy.sh [hunk 2]
```diff
@@ -31,10 +31,29 @@ function main() {
...
+        kubectl rollout status daemonset kepler-exporter -n kepler --timeout 60s
+        #check rollout status
+        if [ $? -eq 0 ]
+        then
+            deploy_status=0
+            break
+        fi
+    done 
+    # if deployment in error
+    if test $[deploy_status] -eq 1
```
Comment (marceloamaral) [inline hack/cluster-deploy.sh:50]: Why don't use the TRAP command?
So, if the error happens in any other place in the script, we will dump the logs

Comment (SamYuan1990) [reply hack/cluster-deploy.sh:50]: may I have a link for TRAP command for details?

File: hack/cluster-deploy.sh (hunk 1 above)
Comment (SamYuan1990) [reply hack/cluster-deploy.sh:38]: will try to see any difference.

File: hack/cluster-deploy.sh (hunk 2 above)
Comment (marceloamaral) [reply hack/cluster-deploy.sh:50]: --> [https://medium.com/@dirk.avery/the-bash-trap-trap-ce6083f36700](https://medium.com/@dirk.avery/the-bash-trap-trap-ce6083f36700)

File: hack/cluster-deploy.sh (hunk 1 above)
Comment (marceloamaral) [reply hack/cluster-deploy.sh:38]: The different will be a more clean code :)

**Pull request merged**